src/lib/components/Dialog.tsx

```diff
@@ -40,7 +40,7 @@ export function Provider({ value, children }: ProviderProps) {
     }
   }, [active])
   return (
-    <div ref={ref}>
+    <div ref={ref} style={{ isolation: 'isolate' }}>
```

Comment:
is this necessary? Looks like it's usually used for opacity/z-index stuff

Reply:
yes - without this, the button's click-target is behind the modal (which is not rendered).

Reply:
Documented this.